app/controllers/triannon/auth_controller.rb

```diff
@@ -0,0 +1,353 @@
+require_dependency "triannon/application_controller"
+
+module Triannon
+  # Adapted from http://image-auth.iiif.io/api/image/2.1/authentication.html
+  class AuthController < ApplicationController
+    include RdfResponseFormats
+
+    AUTH_EXPIRY  =   60 # seconds
+    TOKEN_EXPIRY = 3600 # seconds
+
+    # OPTIONS /auth/login
+    def options
+      # The request MUST use HTTP OPTIONS
+      unless request.options?
+        response.headers.merge!(Allow: 'GET, OPTIONS')
+        err = {
+          error: "invalidRequest",
+          errorDescription: "/auth/login accepts GET or OPTIONS requests",
+          errorUri: "http://image-auth.iiif.io/api/image/2.1/authentication.html"
+        }
+        render_json(err, 405)
+      else
+        if cookies[:login_user]
+          info = service_info_logout
+        else
+          info = service_info_login
+        end
+        # TODO: include optional info, such as service_info_client_identity
+        render_json(info, 200)
+      end
+    end
+
+    # GET /auth/login
+    # HTTP basic authentication
+    # http://image-auth.iiif.io/api/image/2.1/authentication.html#login-service
+    def login
+      # The service must set a Cookie for the Access Token Service to retrieve
+      # to determine the user information provided by the authentication system.
+      if user = authenticate_with_http_basic { |u, p| authorized_user(u, p) }
+        cookies[:login_user] = user
+        session[:login_user] = user
+        redirect_to root_url, notice: 'Successfully logged in.'
+      else
+        respond_to do |format|
+          format.html {
+            request_http_basic_authentication
+          }
+          format.json {
+            err = {
+              error: '401 Unauthorized',
+              errorDescription: 'invalid login details received',
+              errorUri: 'http://image-auth.iiif.io/api/image/2.1/authentication.html'
+            }
+            render_json(err, 401)
+          }
+        end
+      end
+    end
+
+    # GET /auth/logout
+    # http://image-auth.iiif.io/api/image/2.1/authentication.html#logout-service
+    def logout
+      cookies.delete(:login_user)
+      reset_session
+      redirect_to root_url, notice: 'Successfully logged out.'
+    end
+
+    # POST /auth/client_identity
+    # http://image-auth.iiif.io/api/image/2.1/authentication.html#client-identity-service
+    # http://image-auth.iiif.io/api/image/2.1/authentication.html#error-conditions
+    # return json body [String] containing: { "authorizationCode": code }
+    def client_identity
+      unless request.post?
```

Comment:
Sticking to what other developers will expect is the best, IMO.  My understanding is that routes is where those things would normally be, so if (and I have no idea why) anyone would need to find where the handling was, that's probably where they'd look first.  If there's no possibility for the code to be called ... then I'd delete it as unnecessary and just adding extra maintenance.